Scripts/Proxies/ProxyAnimator.cs

```diff
@@ -27,26 +28,53 @@ public void Apply(Transform transform)
     [SerializeField]
     ActionMap m_ProxyActionMap;
 
-    ProxyHelper.ButtonObject[] m_Buttons;
+    [Header("Optional")]
+    [SerializeField]
+    ProxyAffordanceMap m_AffordanceMapOverride;
+
+    Affordance[] m_Affordances;
+    AffordanceDefinition[] m_AffordanceDefinitions;
     InputControl[] m_Controls;
 
     readonly Dictionary<Transform, TransformInfo> m_TransformInfos = new Dictionary<Transform, TransformInfo>();
 
+    bool m_RightHandedProxy;
+    Node m_Node;
+
     public ActionMap actionMap { get { return m_ProxyActionMap; } }
     public bool ignoreLocking { get { return true; } }
-    internal event Action<ProxyHelper.ButtonObject[], Dictionary<Transform, TransformInfo>, ActionMapInput> postAnimate;
 
-    void Start()
+    public Node node
+    {
+        get
+        {
+            return m_Node;
+        }
+
+        set
+        {
+            m_Node = value; m_RightHandedProxy = m_Node == Node.RightHand;
+        }
+    }
+
+    internal event Action<Affordance[], AffordanceDefinition[], Dictionary<Transform, TransformInfo>, ActionMapInput> postAnimate;
+
+    public void Setup(ProxyAffordanceMap affordanceMap, Affordance[] affordances)
     {
-        m_Buttons = GetComponent<ProxyHelper>().buttons;
+        // Assign the ProxyHelper's default AffordanceMap, if no override map was assigned to this ProxyAnimator
+        if (m_AffordanceMapOverride == null)
+            m_AffordanceMapOverride = affordanceMap;
+
+        m_Affordances = affordances;
+        m_AffordanceDefinitions = m_AffordanceMapOverride.AffordanceDefinitions;
     }
 
     public void ProcessInput(ActionMapInput input, ConsumeControlDelegate consumeControl)
     {
-        if (m_Buttons == null)
+        if (m_Affordances == null)
```

Comment:
Probably not necessary after we combine ProxyUI and ProxyHelper

Reply:
I'll leave it for now, till that merge happens (likely after this PR).